### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by carmina-c**

### CODE REVIEW

File: src/auth/auth.guard.ts [hunk 1]
```diff
@@ -7,7 +7,10 @@ import { User } from "src/user/user.entity";
 @Injectable()
 export class JwtAuthGuard extends AuthGuard("jwt") implements IAuthGuard {
 
-	public handleRequest(err: unknown, user: User): any {
+	public handleRequest(err: unknown, user: User, info: Error): any {
```
Comment (danielursu) [inline src/auth/auth.guard.ts]: what is err and info? :))

File: src/auth/auth.guard.ts [hunk 2]
```diff
@@ -7,7 +7,10 @@ import { User } from "src/user/user.entity";
...
+		if (err || info || !user) {
+			throw err || info || new UnauthorizedException()
```
Comment (danielursu) [inline src/auth/auth.guard.ts]:
```suggestion
			throw err || info || new UnauthorizedException();
```

File: src/auth/auth.helper.ts [hunk 3]
```diff
@@ -20,7 +20,9 @@ export class AuthHelper {
 	}
 
 	public async validateUser(decoded: any): Promise<User> {
-  		return this.userRepository.findOne(decoded.id);
+  		return this.userRepository.findOne({
+			where: { id: decoded.id } }
+		);
```
Comment (danielursu) [inline src/auth/auth.helper.ts:25]: is indentation ok?
or Github has an issue

File: src/task/task.controller.ts [hunk 4]
```diff
@@ -12,27 +14,39 @@ export class TaskController {
 	) { }
 
 	@Post()
-	public addTask(@Body() taskDTO: TaskDTO): Promise<InsertResult> {
-		return this.taskService.addTask(taskDTO);
+	@UseGuards(JwtAuthGuard)
```
Comment (danielursu) [inline src/task/task.controller.ts]: try to add this guard under controller decorator
@Controller("task")
@UseGuards(JwtAuthGuard)

File: src/task/task.service.ts [hunk 5]
```diff
@@ -1,66 +1,100 @@
...
 
-	public async updateTaskById(taskDTO: TaskDTO): Promise<TaskDTO> {
-		const task: Task = await this.taskMapper.toEntity(taskDTO);
-		await this.taskRepo.update(taskDTO.id, task);
-		const taskUpdated: Task = await this.taskRepo.findOne({ where: { id: taskDTO.id } });
-		return this.taskMapper.toDTO(taskUpdated);
+	public async updateTaskById(taskDTO: TaskDTO, userId: string): Promise<TaskDTO> {
+		const userDto: UserDTO = await this.userService.getUserById(Number(userId));
+		const user: User = await this.userMapper.toEntity(userDto);
+		if(taskDTO.user === user.id) {
```
Comment (danielursu) [inline src/task/task.service.ts]:
```suggestion
		if (taskDTO.userId === user.id) {
```
we should rename the property to userId

**Review by danielursu — CHANGES_REQUESTED** [review]

File: src/auth/auth.guard.ts (hunk 1 above)
Comment (carmina-c) [reply src/auth/auth.guard.ts]: asa era semnatura metodei :))  handleRequest<TUser = any>(err: any, user: any, info: any, context: ExecutionContext, status?: any): TUser;

File: src/auth/auth.helper.ts (hunk 3 above)
Comment (carmina-c) [reply src/auth/auth.helper.ts:25]: da, arata ok image

**Review by danielursu — APPROVED** [review]

**Pull request merged**